flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSource.java

```diff
@@ -137,6 +148,11 @@
     /** Flag to determine source mode. In upsert mode, it will keep the tombstone message. * */
     protected final boolean upsertMode;
 
+    /** Whether to use legacy {@link FlinkKafkaConsumer} for runtime execution. */
+    protected final boolean fallbackToFlinkKafkaConsumer;
```

Comment:
for the sink we decided against a legacy switch. but maybe it makes sense here if the code duplication and differences to the legacy one are low

Reply:
I think most of the code can be reused for the legacy switch, so I'd like to keep it for now.

Reply:
Sorry for changing my mind in last minutes. I removed this switch in order to be aligned with sink side. Having the "kafka-legacy" for source but not sink looks weird.